### PR TITLE
Getting Started With Debugging In Android Using Chucker

### LINKED ISSUE #5579: [Android]Getting Started With Debugging In Android Using Chucker
### Topic Suggestion
[Android]Getting Started With Debugging In Android Using Chucker

#### Proposed title of article
Getting Started With Debugging In Android Using Chucker

#### Proposed article introduction
Chucker simplifies the inspection of HTTP(S) requests/responses fired by your Android App. Chucker works as an OkHttp Interceptor persisting all those events inside your application, and providing a UI for inspecting and sharing their content.

Apps using Chucker will display a push notification showing a summary of ongoing HTTP activity. Tapping on the notification launches the full Chucker UI. Apps can optionally suppress the notification, and launch the Chucker UI directly from within their own interface.

#### Key takeaways 
- What is Chucker and its applications in Android development.
- How Chucker simplifies the inspection of HTTP(S) requests/responses fired by your Android App.
- How to display a push notification when using Chucker.
- Advantages of using Chucker.

#### Article quality
This article is going to be unique since am going to use MVVM pattern to illustrate the concept of Chucker debugging. I am also going to make it beginner friendly to allow easy understanding. I am also going to show a push notification which is an added advantage to the reader. Although there are certain sites about Chucker, there are still problems with developers understanding the concept.

#### References
[Chucker](https://github.com/ChuckerTeam/chucker)

### PR DESCRIPTION
**Pull request by Hepatrique**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5579

### CODE REVIEW

**Comment by Kayere** [comment]
Hello @Hepatrique,

 Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.

Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.

Be sure to see our resources page to see more info on plagiarism and what is considered as such.

[Getting Started With Debugging In Android Using Chucker #6673 Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/8087487/Getting.Started.With.Debugging.In.Android.Using.Chucker.6673.Copyleaks.Report.pdf)

**Comment by Dawe-7** [comment]
@Kayere, can we run another plagiarism check, thanks.

**Comment by Hepatrique** [comment]
Hello @WanjaMIKE, and @Dawe-7  I think the flags are raised on the code which I cannot change then it is also below 10% as per the rule.

**Comment by Dawe-7** [comment]
> Hello @WanjaMIKE, and @Dawe-7 I think the flags are raised on the code which I cannot change then it is also below 10% as per the rule.

@WanjaMIKE, is this article ready for review or do we need to run another plagiarism check?

**Comment by WanjaMIKE** [comment]
@Hepatrique could you correct the few sentences showing some plagiarism. I shared the report. You can leave the boilerplate code. Thanks

**Comment by Hepatrique** [comment]
Hello @WanjaMIKE , and @Dawe-7 , I have done the changes as suggested and I now hope the article is ready for review.

**Comment by WanjaMIKE** [comment]
I think, they are enough. You can proceed with the review. Thanks

File: content/articles/debugging-with-chucker/index.md [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+    //Coroutines
+    implementation 'org.jetbrains.kotlinx:kotlinx-coroutines-android:1.5.2'
+```
+
+Chucker has two libraries i.e `com.github.chuckerteam.chucker:library` and `com.github.chuckerteam.chucker:library-no-op`. The latter is used to isolate Chucker from the release build. 
+
+This means if you fail to add the second library, Chucker will still work but it will be present in your release build. This is not encouraged due to some reasons that we will discuss below.
+
+Also, ensure Java 8 support is enabled by ensuring that the code below is present in your app-level `build.gradle` file inside the `android` block.
```
Comment (Dawe-7) [inline content/articles/debugging-with-chucker/index.md:83]: Which code are you referring to?

File: content/articles/debugging-with-chucker/index.md [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+In the dependencies, we have added the OkHttp library because Chucker works with its interceptor. The coroutines library is added because OkHttp involves network calls.
```
Comment (Dawe-7) [inline content/articles/debugging-with-chucker/index.md:85]: Does the OkHttp `involves` network calls or `invokes`?

**Review by Dawe-7 — CHANGES_REQUESTED** [review]
Nice article @Hepatrique . Make the suggestions below. Also, ensure your images are of required size 600x400 pixels. Make sure you resize your hero image along with other images used in the article.

Make the changes so we can move on to tech review.

**Comment by Hepatrique** [comment]
Hello @Dawe-7 , I have done the correction, it was a typo error and it ought to be `invokes`.

**Comment by Dawe-7** [comment]
> Hello @Dawe-7 , I have done the correction, it was a typo error and it ought to be `invokes`.

@Hepatrique 

> Hello @Dawe-7 , I have done the correction, it was a typo error and it ought to be `invokes`.

@Hepatrique, kindly ensure you effect all the changes requested.

**Comment by Hepatrique** [comment]
@Dawe-7 , I have done the suggested changes

**Comment by Hepatrique** [comment]
@Dawe-7, kindly proceed to the final review. I have tried resizing the images as suggested. Thank you!!!

**Review by Dawe-7 — APPROVED** [review]
Nice article @Hepatrique. Kindly go through the changes made throughout the review process so as to improve on the flow and format of your future articles.

**Comment by hectorkambow** [comment]
@Hepatrique 

@Dawe-7 

image

**Pull request closed without merging**